fe/fe-core/src/main/java/com/starrocks/scheduler/ExecuteOption.java

```diff
@@ -106,34 +108,32 @@ public void setReplay(boolean replay) {
         isReplay = replay;
     }
 
-    private boolean containsKey(String key) {
-        return taskRunProperties.containsKey(key) && taskRunProperties.get(key) != null;
-    }
-
     /**
-     * If the execute option contains the properties that need to be merged into the task run, eg: it's an internal partition
-     * refresh, needs to merge it into the newer task run.
-     * task in mv refresh
-     * @return
+     * * Check if the current ExecuteOption can be merged with the given option.
+     * Only all taskRunProperties are empty or the same with specific keys, we can merge two pending task runs.
      */
-    public boolean containsToMergeProperties() {
-        if (taskRunProperties == null) {
+    public boolean isMergeableWith(ExecuteOption option) {
```

Comment:
```suggestion
    public boolean isMergeableWith(ExecuteOption other) {
```
`other` will be a better name